engines/dfc_provider/spec/services/authorization_control_spec.rb

```diff
@@ -37,11 +37,39 @@
     end
   end
 
-  def auth(oidc_token:)
+  describe "with OFN API token" do
+    it "finds the user of the API key" do
+      user.update!(spree_api_key: "1234")
+
+      expect(auth(api_token: "1234").user).to eq user
+    end
+
+    it "returns nil if the token doesn't match" do
+      user.update!(spree_api_key: "1234")
+
+      expect(auth(api_token: "123").user).to eq nil
+    end
+
+    it "ignores a missing token" do
+      user.update!(spree_api_key: nil)
+
+      expect(auth(api_token: nil).user).to eq nil
+    end
+
+    it "ignores empty tokens" do
+      user.update!(spree_api_key: "")
+
+      expect(auth(api_token: "").user).to eq nil
+    end
+  end
+
+  def auth(oidc_token: nil, api_token: nil)
+    headers = {}
+    headers["Authorization"] = "Bearer #{oidc_token}" if oidc_token
+    headers["X-Api-Token"] = api_token if api_token
```

Comment:
I was thinking it would be better practice to explicitly check for `.present?` rather than relying on the truthy/falsey interpretation, which can sometimes be unexpected (it varies in different languages).

But it seems it's a pretty simple rule in Ruby: https://www.freecodecamp.org/news/data-types-in-ruby-true-false-and-nil-explained-with-examples
> In Ruby, only false and nil are “falsey,” everything else is “truthy.”